rfc/src/rfcs/0009-function-contracts.md

```diff
@@ -0,0 +1,770 @@
+     will later be replacing and lastly.
+  2. **Check contracts with a `#[proof]` harness.** This would have used e.g. a
+     `#[for_contract]` attributes on a `#[proof]`. Since `#[for_contract]` is
+     *only* valid on a proof, we decided to just imply it and save the user some
+     headache. Contract checking harnesses are not meant to be reused for other
+     purposes anyway and if the user *really* wants to the can just factor out the
+     actual contents of the harness to reuse it.
+
+## Open questions
+
+<!-- For Developers -->
+
+- Is it really correct to return `kani::any()` from the replacement copy, even
+  if it can be a pointer?
+- Making result special. Should we use special syntax here like `@result` or
+  `kani::result()`, though with the latter I worry that people may get confused
+  because it is syntactic and not subject to usual `use` renaming and import
+  semantics. Alternatively we can let the user pick the name with an additional
+  argument to `ensures`, e.g. `ensures(my_result_var, CONDITION)`
+
+  Similar concers apply to `old`, which may be more appropriate to be special
+  syntax, e.g. `@old`.
+
+  See [#2597](https://github.com/model-checking/kani/issues/2597)
+- How to check the right contracts at the right time. By default `kani` and
+  `cargo kani` check all contracts in a file/workspace. This represents the
+  safest option for the user but may be too costly in some cases.
+
+  The user should be provided with options to disable contract checking for the
+  sake of efficiency. Such options may look like this:
+
+  - **By default** (`kani`/`cargo kani`) all local contracts are checked,
+    harnesses are only checked if the contracts they depend on succeeded their check.
+  - **With harness selection** (`--harness`) only those contracts which the
+    selected harnesses depend on are checked.
+  - **For high assurance** passing a `--paranoid` flag also checks contracts for
+    dependencies (other crates) when they are used in stubs.
+  - **Per harness** the users can disable the checking for specific contracts
+    via attribute, like `#[stub_verified(TARGET, trusted)]` or
+    `#[stub_unverified(TARGET)]`. This also plays nicely with `cfg_attr`.
+  - **On the command line** users can similarly disable contract checks by
+    passing (multiple times) `--trusted TARGET` to skip checking those
+    contracts.
+  - **The bold** (or naïve) user can skip all contracts with `--all-trusted`.
+  - **For the lawyer** that is only interested in checking contracts and nothing
+    else a `--litigate` flag checks only contract harnesses.
+
+  Aside: I'm obviously having some fun here with the names, happy to change,
+  it's really just about the semantics.
+ 
+<!-- 
+- Is there any part of the design that you expect to resolve through the RFC process?
+- What kind of user feedback do you expect to gather before stabilization? How will this impact your design? 
+-->
+
+## Future possibilities
```

Comment:
Can you simplify this section a bit? I don't think you need to provide so much details for things that are out of the scope of this RFC.

There are some good information here though. Maybe you can create feature requests for them and move the details to those issues instead.

Reply:
I was looking at our RFC template, and this level of details is welcome. So I'm resolving this comment.